### PR TITLE
StorageService implements IDisposable.

### PR DESCRIPTION
**Pull request by chrisdunelm**
Updates #407

### CODE REVIEW

**Comment by chrisdunelm** [comment]
Once we're happy with the code and xmldoc, I'll do the other REST-based services.

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClient.cs [hunk 1]
```diff
@@ -37,8 +37,14 @@ namespace Google.Cloud.Storage.V1
     /// All instance methods declared in this class are virtual, with an implementation which simply
     /// throws <see cref="NotImplementedException"/>. All these methods are overridden in <see cref="StorageClientImpl"/>.
     /// </para>
+    /// <para>
+    /// This class implements <see cref="IDisposable"/>; however, the <see cref="Dispose"/> method only requires calling
+    /// if many short-lived instances of <see cref="StorageClient"/> are being created.
+    /// Most code would be expected to create a single <c>StorageClient</c> instance, and use this instance throughout
+    /// the lifetime of the application. In this case, <c>Dispose</c> need not be called.
```
Comment (jskeet) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClient.cs]: Even creating multiple clients isn't a problem in many cases, of course. We'll definitely want to have a broader FAQ entry about this, but we can do that later.

**Comment by chrisdunelm** [comment]
Yes, I wondered about making Dispose virtual here, with an empty method body; then overriding in the Impl to actual do the disposal.

**Comment by jskeet** [comment]
Empty rather than throwing NotImplementedException? Not sure which I prefer.

**Comment by chrisdunelm** [comment]
Empty means that an alternative implementation (possibly for testing) doesn't need to worry about disposal if disposal isn't required; otherwise code outside of that users control may call Dispose which will throw the exception.

**Comment by jskeet** [comment]
Sure - it's easy enough to override where appropriate. Let's go for that.

**Comment by chrisdunelm** [comment]
I'll merge on green

**Pull request merged**